openapi_python_client/parser/properties/__init__.py

```diff
@@ -388,14 +405,15 @@ def _property_from_data(
     """ Generate a Property from the OpenAPI dictionary representation of it """
     name = utils.remove_string_escapes(name)
     if isinstance(data, oai.Reference):
-        reference = Reference.from_ref(data.ref)
-        existing = schemas.enums.get(reference.class_name) or schemas.models.get(reference.class_name)
-        if existing:
-            return (
-                attr.evolve(existing, required=required, name=name),
-                schemas,
+        return _property_from_ref(name=name, required=required, nullable=False, data=data, schemas=schemas)
+
+    for attribute in ["allOf", "anyOf", "oneOf"]:
+        sub_data = getattr(data, attribute)
+        if sub_data and len(sub_data) == 1 and isinstance(sub_data[0], oai.Reference):
+            return _property_from_ref(
+                name=name, required=required, nullable=data.nullable, data=sub_data[0], schemas=schemas
```

Comment:
I'm probably just misunderstanding something, but what happens if there's more than one thing in an `allOf`/`anyOf`/`oneOf` list?

Reply:
In that case, `len(sub_data) > 1` and it would fall through to the pre-existing implementation below. For `anyOf` and `oneOf` that is L421, for `allOf` that is L457.

Reply:
Ah yep I see that now - thats what I get for missing the boat on reviewing the original `allOf`/`anyOf`/`oneOf` ticket 😄